elements/howto-tree/howto-tree.js

```diff
@@ -0,0 +1,494 @@
+ * `aria-activedescendant` and the `id` of the currently active child. The
+ * effect is similar to using roving tabindex, and is presented in this case to
+ * show an alternative approach to indicating active children.
+ */
+
+(function() {
+  /**
+   * Define keycodes to help with handling keyboard events.
+   */
+  const KEYCODE = {
+    UP: 38,
+    DOWN: 40,
+    LEFT: 37,
+    RIGHT: 39,
+    HOME: 36,
+    END: 35,
+    SPACE: 32,
+    ENTER: 13,
+  };
+
+  /**
+   * A helper to quickly identify `HowToTreeItem`/`HowToTreeItemGroup` nodes.
+   * Because both `HowToTreeItem` and `HowToTreeItemGroup` elements can have
+   * a `role=treeitem`, this helper will match against either.
+   */
+  function isTreeItem(node) {
+    return node.nodeName.toLowerCase() === 'howto-tree-item' ||
+           node.nodeName.toLowerCase() === 'howto-tree-item-group';
+  }
+
+  /**
+   * A helper to specifically identify `HowToTreeItemGroup` elements.
+   */
+  function isTreeItemGroup(node) {
+    return node.nodeName.toLowerCase() === 'howto-tree-item-group';
+  }
+
+  // `HowToTreeItemCounter` counts the number of `treeItem` instances
+  // created. The number is used to generated new, unique `id`s.
+  let HowToTreeItemCounter = 0;
+
+  class HowToTreeItem extends HTMLElement {
+    static get observedAttributes() {
+      return ['selected'];
+    }
+
+    connectedCallback() {
+      if (!this.hasAttribute('role'))
+        this.setAttribute('role', 'treeitem');
+
+      // If the element doesn't already have an `id`, generate one for it.
+      // Every node needs an `id` so it can be referenced by
+      // `aria-activedescendant`.
+      if (!this.id)
+        this.id = `howto-tree-item-generated-${HowToTreeItemCounter++}`;
+    }
+
+    /**
+     * Properties and their corresponding attributes should mirror one another.
+     * To this effect, the property setter for `selected` handles truthy/falsey
+     * values and reflects those to the state of the attribute.
+     * It's important to note that there are no side effects taking place in
+     * the property setter. For example, the setter does not set
+     * `aria-selected`.
+     * Instead, that work happens in the `attributeChangedCallback`.
+     * As a general rule, make property setters very dumb, and if setting a
+     * property or attribute should cause a side effect (like setting a
+     * corresponding ARIA attribute) do that work in the
+     * `attributeChangedCallback`. This will avoid having to manage complex
+     * attribute/property reentrancy scenarios.
+     */
+    set selected(value) {
+      const isSelected = Boolean(value);
+      if (isSelected)
+        this.setAttribute('selected', '');
+      else
+        this.removeAttribute('selected');
+    }
+
+    get selected() {
+      return this.hasAttribute('selected');
+    }
+
+    attributeChangedCallback(name, value) {
+      if (name === 'selected')
+        this.setAttribute('aria-selected', this.selected);
```

Comment:
Maybe a comment explaining why you're setting this here, analogous to the explanation about `aria-expanded`?